npk/src/manifest.rs

```diff
@@ -447,10 +448,18 @@ impl Manifest {
         build: vec![],
     });
 
-    fn verify(&self) -> Result<(), ManifestError> {
+    pub fn from_yaml<R: io::Read>(reader: R) -> Result<Self, Error> {
+        serde_yaml::from_reader(reader).map_err(Error::SerdeYaml)
+    }
+
+    pub fn to_yaml<W: io::Write>(&self, writer: W) -> Result<(), Error> {
+        serde_yaml::to_writer(writer, self).map_err(Error::SerdeYaml)
+    }
```

Comment:
I'm struggling with the names. `from_yaml` seems to me more like a `from_reader`.